pkg/gardenlet/controller/shoot/shoot_control_migrate.go

```diff
@@ -126,7 +126,8 @@ func (c *Controller) runPrepareShootControlPlaneMigration(o *operation.Operation
 			return nil
 		}),
 		utilerrors.ToExecute("Retrieve the Shoot namespace in the Seed cluster", func() error {
-			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, namespace); err != nil {
+			botanist.SeedNamespaceObject = &corev1.Namespace{}
+			if err := botanist.K8sSeedClient.Client().Get(ctx, client.ObjectKey{Name: o.Shoot.SeedNamespace}, botanist.SeedNamespaceObject); err != nil {
```

Comment:
after this if block, you need something like 
```golang
namespace = botanist.SeedNamespaceObject
```

Reply:
I'll just get rid of the `namespace` var and use
```
nonTerminatingNamespace = botanist.SeedNamespaceObject.Status.Phase != corev1.NamespaceTerminating
```

seems like the better approach